### ISSUE TITLE
--threads=max should be default, except in CI

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#2795]
Why?

Because faster Psalm is good.

**Comment by iluuu1994** [comment]
Why not in CI? As long as the pcntl extension is enabled (which admittedly is probably not very often).

**Comment by muglug** [comment]
> Why not in CI?

Because I've encountered (anecdotally) instances of where threading doesn't perform optimally on resource-constrained boxes in CI – maybe the machine has the capability, but it's performing a ton of other tasks at the same time.

**Comment by iluuu1994** [comment]
Ok, I was just interested.

**Comment by jarstelfox** [comment]
@muglug, This is just an edge-case I noticed, but worth mentioning.

Back around psalm 8.7.2, we ran psalm with 4 threads. If a thread, hit an internal expectation from psalm, all the classes it would have analyzed were never seen by psalm. worse yet: Psalm report it passed when one of the threads had crashed.

We found it possible to make psalm errors (which were not reported) in the files that were not looked at by the analyzer.

Since you would be making multi-threaded the default, I thought it was worth mentioning.

I'm not sure if you have fixed this or I should move this to a new issue for you.
